activerecord/test/cases/connection_adapters/connection_handler_test.rb

```diff
@@ -275,6 +275,42 @@ def test_connection_specification_name_should_fallback_to_parent
         ActiveRecord::Base.connection_specification_name = "ActiveRecord::Base"
       end
 
+      unless in_memory_db?
+        def test_connects_to_on_application_record_sets_connection_specification_name_on_active_record_base
```

Comment:
The test name doesn't seem to reflect what the test is testing for here. From the name, it seems like we're expecting `ActiveRecord::Base.connection_specification_name` to be set when `ApplicationRecord.connects_to` is called. In the test case itself, `ActiveRecord::Base.connection_specification_name` is set manually. Should the test name be something like the following instead?

```suggestion
        def test_connects_to_on_application_record_sets_connection_specification_name_of_active_record_base
```

Reply:
> From the name, it seems like we're expecting `ActiveRecord::Base.connection_specification_name` to be set when `ApplicationRecord.connects_to` is called.

That is what's being tested; or rather, that it _doesn't_ set `ApplicationRecord.connection_specification_name`.

> In the test case itself, `ActiveRecord::Base.connection_specification_name` is set manually.

That's part of the assert phase, since it's hard to verify the behaviour directly through public API. If we didn't care about that, the test could look like this, without touching `ActiveRecord::Base.connection_specification_name`:

```ruby
ApplicationRecord.connects_to database: { writing: :primary }
assert_nil ApplicationRecord.instance_variable_get(:@connection_specification_name)
```

Reply:
> that it _doesn't_ set `ApplicationRecord.connection_specification_name`.

O right this makes sense now. Thank you for explaining :+1: